python/ml4ir/base/data/tfrecord_reader.py

```diff
@@ -761,7 +761,7 @@ def read(
 
     # Create BatchedDataSet
     if batch_size:
-        dataset = dataset.batch(batch_size, drop_remainder=True)
+        dataset = dataset.batch(batch_size, drop_remainder=False)
```

Comment:
the gist of this PR.

Reply:
@balikasg I thought that for training, we would still drop the remainder?

Reply:
this is what was hinted by @jakemannix in the slack discussion. But: 
- I could not think of an advantage doing so. 
- Doing so would require an extra argument to be exposed to allow tweaking with this value depending on `fit`/`predict`/`evaluate`. I think this would lead to unnecessary code complexity. 
So i decided to go with the simpler solution. Is there any particular reason you would prefer to have `drop_remainder=True` during training @lastmansleeping ?

Reply:
Not really. I just worried ranking model would not work with `drop_remainder=False` at training time. I've seen this in the past. But happy to see all tests pass.